submissions/nadsatt/dom-api/app.js

```diff
@@ -0,0 +1,192 @@
+        rating: 6,
+        colors: ['rgb(7,229,169)', 'rgb(106, 115, 94)', 'rgb(155, 163, 146)', 'rgb(202, 181, 158)']
+    },
+    {
+        id: 2,
+        name: 'cat of the year', 
+        description: 'Tom Dobbs, the satirical host of a political talk show, decides to shake things up by running for president after making comments that he would be a better president than the leader who currently occupies the White House. His fans begin a grassroots campaign that catapults him into the Oval Office, but soon after moving in it is revealed a computer glitch mistakenly awarded him the presidency. Tom is left with a tough choice, stay the course or head back behind the microphone.',
+        rating: 2,
+        colors: ['rgb(207,163,179)', 'rgb(146,43,62)', 'rgb(23,69,111)', 'rgb(189,200,196)']
+    },
+    {
+        id: 3,
+        name: 'kitty potter', 
+        description: 'Harry Potter\'s third year at Hogwarts starts off badly when he learns deranged killer Sirius Black has escaped from Azkaban prison and is bent on murdering the teenage wizard. While Hermione\'s (Emma Watson) cat torments Ron\'s sickly rat, causing a rift among the trio, a swarm of nasty Dementors is sent to protect the school from Black. A mysterious new teacher helps Harry learn to defend himself, but what is his secret tie to Sirius Black?',
+        rating: 9,
+        colors: ['rgb(0,145,194)', 'rgb(0,26,57)', 'rgb(175, 171, 171)', 'rgb(116,153,156)']
+    },
+    {
+        id: 4,
+        name: 'new meov', 
+        description: 'High-school student Bella Swan, always a bit of a misfit, doesn\'t expect life to change much when she moves from sunny Arizona to rainy Washington state. Then she meets Edward Cullen, a handsome but mysterious teen whose eyes seem to peer directly into her soul. Edward is a vampire whose family does not drink blood, and Bella, far from being frightened, enters into a dangerous romance with her immortal soulmate.',
+        rating: 5,
+        colors: ['rgb(23,131,72)', 'rgb(93, 150, 18)', 'rgb(175, 171, 171)', 'rgb(128,126,44)']
+    },
+];
+
+headerIcon.onclick = toggleSidebar;
+sidebarIcon.onclick = toggleSidebar;
+
+fillPageByMovieId(0);
+
+function toggleSidebar(){
+    headerIcon.classList.toggle('icon-wrapper--active');
+    sidebarIcon.classList.toggle('icon-wrapper--active');
+    sidebar.classList.toggle('sidebar--active');
+}
+
+function fillPageByMovieId(id){
+    const movie = movies.find(movie => id === movie.id);
+
+    fillSidebarList(movies.map(movie => movie.name));
+    fillPosterSection(movies[id].name);
+    fillRatingSection(movies[id].rating);
+    fillActorsSection(movies[id].name);
+    fillDescriptionSection(movies[id].description);
+    defineColors(movies[id].colors);
+    defineBackgrounds(movies[id].name);
+}
+
+function fillSidebarList(names){
+    if(sidebarList.firstElementChild){
```

Comment:
When you'll move `fillSidebarList` function outside your `fillPageByMovieId` you can get rid of the checking for first element existence.